rpc/src/optimistically_confirmed_bank_tracker.rs

```diff
@@ -259,6 +267,7 @@ impl OptimisticallyConfirmedBankTracker {
         highest_confirmed_slot: &mut Slot,
         newest_root_slot: &mut Slot,
         slot_notification_subscribers: &Option<Arc<RwLock<Vec<SlotNotificationSender>>>>,
+        prioritization_fee_cache: &Arc<PrioritizationFeeCache>,
```

Comment:
I think this should likely be passed as a `&PrioritizationFeeCache` as we are not taking ownership nor cloning.

Reply:
Commenting because there's on-going efforts to be more consistent of how `&Arc` is used. See #32698

Reply:
probably `OptimisticallyConfirmedBankTracker` does need a copy of `PrioritizationFeeCache`, maybe can just pass `Arc<>`. But I was trying to be consistent with `subscriptions: &Arc<RpcSubscriptions>,` few lines above.

Reply:
Most likely RpcSubscriptions handling of Arc also needs to be cleaned up. No need to match a bad pattern 😅

Reply:
only updated the `&Arc` for prioritization_fee_cache here for easier backporting. 

#32729 to remove other `&Arc`

Reply:
@apfitzge if you can review one more time before merge would be great!